packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -95,7 +100,15 @@ export const useExecuteCode: EnsembleActionHook<
           screen,
           js,
           merge(
-            { ensemble: { storage, formatter, env: appContext?.env } },
+            {
+              ensemble: {
+                storage,
+                formatter,
+                env: appContext?.env,
+                navigateScreen: (targetScreen: NavigateScreenAction) =>
+                  navigateApi(targetScreen, screen, customScope, storage),
```

Comment:
Where is the location API?

Reply:
I supposed this navigateScreen is alias of location api, for move from one page to other page inside the useExecuteCode API

Reply:
The location and history APIs should be separate members. We need a way to view the current location and history.

Reply:
okay should I consider this navigate API as a history API ?

Reply:
@evshi As per the react router v6, history is removed from the react router dom. and also we can't get the history stack from the window.history as per the security related concern. because the history is communicate with the tab, not the specific domain.